### DISCUSSION TITLE
Accessibility issues within debugger

### DISCUSSION THREAD

**Discussion by hedgeberg** [NationalSecurityAgency/ghidra#3204] [Debugger]
Folks,

I have an intern who's been using some of the tooling I whipped up to analyze a binary with me in the debugger. He's red/green colorblind, which created a bit of confusion when we were working with breakpoints just now, see:
![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/3290125/125124175-b1bc0300-e0c5-11eb-8926-9c7b9426e74e.png)

He has trouble differentiating between the grey and the red, which appears to be the only way to tell if a saved breakpoint is active in the current session. Is there an easy way to tweak debugger colors currently? If not, it might be worth considering.

**Comment by d-millar** [comment]
@hedgeberg Well, the answer is yes and no, unfortunately. If you go to Edit->Tool Options...->Debugger-> Colors, you can change the color of the background in the listings associated with a breakpoint in various states (enabled, disabled, inactive, etc.).  This does not, however, change the color of the icon proper used in the margin of the listings and in the Breakpoints window.  Right now, to change that you would have to replace the icon itself in Debug Debugger src/main/resources/images, which you could definitely do but obviously is a bit more work.  That said, we certainly get it and will look into the issue.  Not sure we can make the icons customizable, but we could probably figure out an easy way to swap in an alternate color set.

**Reply by hedgeberg** [reply]
Great, thanks so much! An alternate color set is likely a good idea, although I'm not colorblind myself so I can't really advise on how best to approach that. I know Microsoft has done a lot of work on things like high contrast modes for the visually impaired, might be worth looking into? Probably would also be best to get comments from actual colorblind people.

**Reply by astrelsky** [reply]
> @hedgeberg Well, the answer is yes and no, unfortunately. If you go to Edit->Tool Options...->Debugger-> Colors, you can change the color of the background in the listings associated with a breakpoint in various states (enabled, disabled, inactive, etc.). This does not, however, change the color of the icon proper used in the margin of the listings and in the Breakpoints window. Right now, to change that you would have to replace the icon itself in Debug Debugger src/main/resources/images, which you could definitely do but obviously is a bit more work. That said, we certainly get it and will look into the issue. Not sure we can make the icons customizable, but we could probably figure out an easy way to swap in an alternate color set.

For my class plugin I wanted to reuse the `GhidraClass` symbol tree icon with a different color but didn't want to copy the image and have them in my repo. I used this little trick with `ColorModel` to circumvent it.

[TypeInfoNode.java#L178-L204](https://github.com/astrelsky/Ghidra-Cpp-Class-Analyzer/blob/4b3fbb44353bf84b7e9b0f8b52c312e098f197dd/src/main/java/cppclassanalyzer/plugin/typemgr/node/TypeInfoNode.java#L178-L204)

The custom `ColorModel` classes are [here](https://github.com/astrelsky/Ghidra-Cpp-Class-Analyzer/tree/master/src/main/java/cppclassanalyzer/plugin/typemgr/icon). It should be really simple to re-implement.

**Comment by d-millar** [comment]
NB: If you do go the changing the background route until we have a better solution, bear in mind clicking the address in the Breakpoints window will take you to that address in the listing, so that you can see the context.  Not ideal, I know, but....

**Comment by dragonmacher** [comment]
We could also update the icons to be visually different more than in just color.  For example, the disabled breakpoint could have have a line through it.

**Comment by d-millar** [comment]
@dragonmacher hadn't thought of that - seems like an obviously good first step now that you mention it

**Comment by d-millar** [comment]
I should check the tooltip too.  I think it’s currently empty, or the lock on restoring the column means it appears empty.

**Comment by ryanmkurtz** [comment]
Fixed by f97ced9d50ed1a2cc3eedb76cd99618a0fb0d45e
